libs/eavmlib/src/emscripten.erl

```diff
@@ -895,85 +1231,146 @@ register_touchstart_callback(_Target, _Options) ->
 %% @see register_touchstart_callback/2
--spec register_touchmove_callback(html5_target(), register_options(), any()) -> register_result().
+%% @param Target target to register keypress on
+%% @param Options options for event handling
+%% @param UserData user data passed back
+%% @returns Result is a `register_result()'
+%% @end
+-spec register_touchmove_callback(
+    Target :: html5_target(), Options :: register_options(), UserData :: any()
+) -> Result :: register_result().
 register_touchmove_callback(_Target, _Options, _UserData) ->
     erlang:nif_error(undefined).
 
 %% @doc Unregister a touchmove event handler.
 %% @see unregister_touchstart_callback/1
--spec unregister_touchmove_callback(html5_target() | listener_handle()) ->
+%% @param TargetOrHandle Target or handle
+%% @returns ok or `{error, Reason}'
+%% @end
+-spec unregister_touchmove_callback(TargetOrHandle :: html5_target() | listener_handle()) ->
     ok | {error, register_error_reason()}.
 unregister_touchmove_callback(_TargetOrHandle) ->
     erlang:nif_error(undefined).
 
-%% @equiv register_touchcancel_callback(_Target, [])
--spec register_touchcancel_callback(html5_target()) -> register_result().
+%% @doc Register for touchcancel events.
+%% @equiv register_touchcancel_callback(Target, [])
+%% @end
+-spec register_touchcancel_callback(Target :: html5_target()) -> Result :: register_result().
 register_touchcancel_callback(_Target) ->
     erlang:nif_error(undefined).
 
 %% @doc Register for touchcancel events.
 %% @see register_touchstart_callback/2
--spec register_touchcancel_callback(html5_target(), register_options()) -> register_result().
+%% @param Target target to register keypress on
+%% @param Options options for event handling
+%% @returns Result is a `register_result()'
+%% @end
+-spec register_touchcancel_callback(Target :: html5_target(), Options :: register_options()) ->
+    Result :: register_result().
 register_touchcancel_callback(_Target, _Options) ->
     erlang:nif_error(undefined).
 
 %% @doc Register for touchcancel events.
 %% @see register_touchstart_callback/2
--spec register_touchcancel_callback(html5_target(), register_options(), any()) -> register_result().
+%% @param Target target to register keypress on
+%% @param Options options for event handling
+%% @param UserData user data passed back
+%% @returns Result is a `register_result()'
+%% @end
+-spec register_touchcancel_callback(
+    Target :: html5_target(), Options :: register_options(), UserData :: any()
+) -> Result :: register_result().
 register_touchcancel_callback(_Target, _Options, _UserData) ->
     erlang:nif_error(undefined).
 
 %% @doc Unregister a touchcancel event handler.
 %% @see unregister_touchstart_callback/1
--spec unregister_touchcancel_callback(html5_target() | listener_handle()) ->
+%% @param TargetOrHandle Target or handle
+%% @returns ok or `{error, Reason}'
+%% @end
+-spec unregister_touchcancel_callback(TargetOrHandle :: html5_target() | listener_handle()) ->
```

Comment:
is changing spec from `unregister_touchcancel_callback(html5_target() | listener_handle())` to `unregister_touchcancel_callback(TargetOrHandle :: html5_target() | listener_handle())` a requirement?

Reply:
Not including the argument name in in the spec (i.e. only the type) prevents the argument from being displayed in the documentation.